client/src/client/session_manager_client.py

```diff
@@ -144,8 +134,10 @@ async def _run_server_handshake(
             public_key_bundle,
         ) in active_friends_message.payload["friends"].items():
             identity_key, one_time_key = public_key_bundle
-            self.potential_sessions[pub_key_digest] = PotentialSession(
-                identity_key=identity_key, one_time_key=one_time_key
+            self.state_machine.add_potential_session(
```

Comment:
I think it would be better if that attribute was called session_sm or something that would bring info that it is about sessions. State Machine directly connects in my mind to https://en.wikipedia.org/wiki/Finite-state_machine